pkg/segment/query/processor/sortcommand_test.go

```diff
@@ -278,3 +278,107 @@ func Test_SortCommand_withRRCs(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, expectedCol2, actualCol2)
 }
+
+func TestSortMultipleDataTypes(t *testing.T) {
+	knownValues := map[string][]utils.CValueEnclosure{
+		"col1": {
+			{Dtype: utils.SS_DT_STRING, CVal: "z"},
+			{Dtype: utils.SS_DT_STRING, CVal: "a"},
+			{Dtype: utils.SS_DT_BACKFILL, CVal: nil},
+			{Dtype: utils.SS_DT_FLOAT, CVal: float64(3)},
+			{Dtype: utils.SS_DT_STRING, CVal: "2"},
+			{Dtype: utils.SS_DT_STRING, CVal: "1"},
+			{Dtype: utils.SS_DT_BACKFILL, CVal: nil},
+			{Dtype: utils.SS_DT_UNSIGNED_NUM, CVal: uint64(5)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(4)},
+			{Dtype: utils.SS_DT_BOOL, CVal: true},
+			{Dtype: utils.SS_DT_BOOL, CVal: false},
+			{Dtype: utils.SS_DT_BOOL, CVal: false},
+			{Dtype: utils.SS_DT_BOOL, CVal: true},
+			{Dtype: utils.SS_DT_STRING, CVal: "A"},
+			{Dtype: utils.SS_DT_STRING, CVal: "Z"},
+		},
+	}
+	iqr1 := iqr.NewIQR(0)
+	err := iqr1.AppendKnownValues(knownValues)
+	assert.NoError(t, err)
+
+	// Test with ascending order.
+	sorter := &sortProcessor{
+		options: &structs.SortExpr{
+			SortEles: []*structs.SortElement{
+				{Field: "col1", SortByAsc: true, Op: ""},
+			},
+			Limit: 100,
+		},
+	}
+
+	expected := []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "1"},
+		{Dtype: utils.SS_DT_STRING, CVal: "2"},
+		{Dtype: utils.SS_DT_FLOAT, CVal: float64(3)},
+		{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(4)},
+		{Dtype: utils.SS_DT_UNSIGNED_NUM, CVal: uint64(5)},
+		{Dtype: utils.SS_DT_STRING, CVal: "A"},
```

Comment:
nit: add a two-digit string number to this test (e.g., would `"15"` be between `"1"` and `"2"` or after `5`?)

Are we handling all these mixed types (string, int, bool), the same way Splunk does?

Reply:
addressed in the latest commit. Yes, its similar to what they do.